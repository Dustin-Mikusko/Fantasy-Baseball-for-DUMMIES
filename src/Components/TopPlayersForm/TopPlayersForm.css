.stat-form {
  margin-top: 150px;
  border: none;
  height: 40px;
}

.stat-select, .season-select {
  height: 53px;
  width: 200px;
  margin-right: 10px;
  border: 3px solid #041F42;
  font-size: .8em;
}

.top-players-btn {
  background-color: #041F42;
  color: white;
  font-size: 1.2em;
  border: none;
  border-radius: 3px;
  width: 90px;
  height: 45px;
}

.top-players-btn:hover {
  background-color: rgb(6, 71, 155);
  cursor: pointer;
}

.top-players-btn:active {
  transform: scale(.97)
}


