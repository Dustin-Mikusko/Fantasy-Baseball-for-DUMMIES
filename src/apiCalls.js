export const fetchSingleName = name => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  const urlName = `${name}%25`;
  return fetch(`http://lookup-service-prod.mlb.com/json/named.search_player_all.bam?sport_code='mlb'&active_sw='Y'&name_part='${urlName}'`, options)
    .then(res => {
      if (!res.ok) {
        console.log('oh hi')
        throw Error('Uh oh! Player not found or player is no longer active')
      }
      console.log('oh hello')
      return res.json();
    })
}

export const fetchFullName = name => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  return fetch(`http://lookup-service-prod.mlb.com/json/named.search_player_all.bam?sport_code='mlb'&active_sw='Y'&name_part='${name}'`, options)
    .then(res => {
      if (!res.ok) {
        throw Error('Uh Oh! Player not found or player is no longer active')
      }
      return res.json();
    })
}

export const fetchSeasonHittingStats = (year, id) => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  return fetch(`http://lookup-service-prod.mlb.com/json/named.sport_hitting_tm.bam?league_list_id='mlb'&game_type='R'&season=${year}&player_id=${id}`, options)
    .then(res => {
      if (!res.ok) {
        throw Error('Uh Oh! Player not found or player is no longer active')
      }
      return res.json();
    })
}

export const fetchTop25 = (year, stat) => {
  const options = {
    "method": "GET",
    "headers": {
      "x-rapidapi-host": "mlb-data.p.rapidapi.com",
      "x-rapidapi-key": "1b5b2533a1msh5cb04fc7ae14a5fp1cb4bdjsnca3f21f97e9d"
    }
  };
  return fetch(`http://lookup-service-prod.mlb.com/json/named.leader_hitting_repeater.bam?sport_code='mlb'&results=25&game_type='R'&season=${year}&sort_column=${stat}&leader_hitting_repeater.col_in=rbi&leader_hitting_repeater.col_in=player_id&leader_hitting_repeater.col_in=name_display_first_last&leader_hitting_repeater.col_in=team_name`, options)
    .then(res => {
        if (!res.ok) {
          throw Error('Opps! There was an error on the play..')
        }
        return res.json();
    })
}


