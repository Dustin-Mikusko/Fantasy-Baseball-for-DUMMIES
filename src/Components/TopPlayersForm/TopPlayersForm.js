import React, { Component } from 'react';
import './TopPlayersForm.css';
import Header from '../Header/Header';
import { fetchTop25 } from '../../apiCalls';
import TopPlayersContainer from '../TopPlayersContainer/TopPlayersContainer'

export class TopPlayersForm extends Component {
  constructor() {
    super();
    this.state = {
      stat: '',
      season: '',
      error: false,
      players: []
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value})
  }

  getTopPlayers = () => {
    fetchTop25(this.state.season, this.state.stat)
      .then(data => {
        console.log(data.leader_hitting_repeater.leader_hitting_mux.queryResults.row);
        this.setState({ players: data.leader_hitting_repeater.leader_hitting_mux.queryResults.row })
      })
      .catch(err => this.setState({ error: true }))
  }

  render() {
    const bigStat = this.state.stat.toUpperCase();
    return (
    <>
      <Header />
      <form className='stat-form'>
        <select className='stat-select' name='stat' onChange={this.handleChange}>
          <option>--Select Stat--</option>
          <option value='bb'>BB</option>
          <option value='rbi'>RBI</option>
          <option value='sb'>SB</option>
          <option value='hr'>HR</option>
          <option value='obp'>OBP</option>
          <option value='r'>R</option>
          <option value='ab'>AB</option>
          <option value='avg'>AVG</option>
          <option value='slg'>SLG</option>
          <option value='ops'>OPS</option>
          <option value='so'>SO</option>
        </select>
        <select className='season-select'name='season' onChange={this.handleChange}>
          <option>--Select Season--</option>
          <option value='2019'>2019</option>
          <option value='2018'>2018</option>
          <option value='2017'>2017</option>
          <option value='2016'>2016</option>
          <option value='2015'>2015</option>
        </select>
        <button type='button' className='top-players-btn' onClick={this.getTopPlayers}>GO</button>
      </form>
      {this.state.error && <p>Oops! Something went wrong..</p>}
      {this.state.players.length && 
        <div className='top-container'>
        <section className='stat-explanation'>
          <h1>{bigStat}:</h1>
          <p>This is an explantaion of the stat.</p>
        </section>
      <TopPlayersContainer 
        players={this.state.players}
        stat={this.state.stat} 
      />
      </div>}
    </>
    )
  }
}

export default TopPlayersForm;
