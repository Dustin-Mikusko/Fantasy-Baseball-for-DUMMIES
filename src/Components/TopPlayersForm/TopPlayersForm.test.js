import React from 'react';
import TopPlayersForm from './TopPlayersForm';
import { shallow } from 'enzyme';
import { fetchTop25 } from '../../apiCalls';


jest.mock('../../apiCalls.js');

describe('TopPlayersForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TopPlayersForm
      
    />);
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleChange', () => {
    it('should update state when handleChange is called', () => {
      const mockState = {
        stat: '',
        season: ''
      };
      const mockEvent = {
        target: {
          name: 'stat',
          value: 'bb'
        }
      };

      wrapper.find('.stat-select').simulate('change', mockEvent);

      expect(wrapper.state('stat')).toEqual('bb');
    });
  })

  describe('getTopPlayers', () => {
    beforeEach(() => {
      fetchTop25.mockImplementation(() => {
        return Promise.resolve({
          ok: true,
          json: () => Promise.resolve({
            "leader_hitting_repeater": {
              "copyRight": " Copyright 2019 MLB Advanced Media, L.P.  Use of any content on this page acknowledges agreement to the terms posted here http://gdx.mlb.com/components/copyright.txt  ",
              "leader_hitting_mux": {
                  "sort_column": "'hr'",
                  "queryResults": {
                      "created": "2020-01-13T03:02:17",
                      "totalSize": "28",
                      "row": [
                          {
                              "hr": "53",
                              "team_name": "New York Mets",
                              "name_display_first_last": "Pete Alonso",
                              "player_id": "624413"
                          }
                      ]
                    }
                  }
                }
            })
          })
      });
    });
    it('should invoke getTopPlayers on button click', () => {
      wrapper.instance().getTopPlayers = jest.fn();
      wrapper.instance().forceUpdate();
      wrapper.find('.top-players-btn').simulate('click');

      expect(wrapper.instance().getTopPlayers).toHaveBeenCalled()
    });

    it('should invoke fetchTop25 when getTopPlayers is fired', () => {
      const mockState = {
        season: '2019',
        stat: 'hr'
      }
      wrapper.setState(mockState);
      wrapper.instance().getTopPlayers();
      expect(fetchTop25).toHaveBeenCalledWith('2019', 'hr');
    });

    it('should set players state with ')
  })
})
