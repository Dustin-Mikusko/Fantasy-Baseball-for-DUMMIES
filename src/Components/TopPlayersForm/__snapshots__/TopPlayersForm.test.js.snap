// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPlayersForm should match the snapshot 1`] = `ShallowWrapper {}`;
